cardinal/server/option.go

```diff
@@ -22,3 +22,34 @@ func DisableSwagger() Option {
 		s.config.isSwaggerDisabled = true
 	}
 }
+
+// DisableReplayProtection disables replay protection, so signed
+// messages can be reused without being signed again, and they never expire.
+// If replay protection is disabled, then WithMessageExpiration and
+// WithHashCacheSize options have no effect
+func DisableReplayProtection() Option {
```

Comment:
There's no instances where we would want signature verification enabled but without replay protection. We should lean towards less options than more because it increases the number of edge cases.

Reply:
Sounds good, I'll fix it.

Reply:
This is fixed by removing DisableReplayProtection(), the flag it set, and replacing it with the sign validation flag wherever it was used.